extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -378,6 +383,16 @@ public void build(ApplicationInfoBuildItem applicationInfo,
                     kubernetesHealthLivenessPath,
                     kubernetesHealthReadinessPath);
 
+            decorators.stream().filter(d -> d.matches(Decorator.class)).forEach(i -> {
```

Comment:
What's the rationale of this change? I think we at least need a comment about it

Reply:
As I didn't felt comfortable is exposing `Decorate` specific classes in the kubernetes spi. Decorators are passed to build items as Objects. So, at some point we need to check if the build item is indeed holding an instance of the `Decorator.class`. 

Definitely not the most elegant solution, but I feel it's somehow better than dragging decorate to the kubernetes spi.
I am open about it though.

Reply:
Understood. I'm fine with it, but let's please add some relevant comments to the source